library/core/src/intrinsics.rs

```diff
@@ -1893,6 +1893,22 @@ extern "rust-intrinsic" {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[rustc_const_unstable(feature = "const_intrinsic_copy", issue = "80697")]
     pub fn copy<T>(src: *const T, dst: *mut T, count: usize);
+
+    /// Determines whether the raw bytes of the two values are equal.
+    ///
+    /// The is particularly handy for arrays, since it allows things like just
+    /// comparing `i96`s instead of forcing `alloca`s for `[6 x i16]`.
+    ///
+    /// Above some backend-decided threshold this will emit calls to `memcmp`,
+    /// like slice equality does, instead of causing massive code size.
+    ///
+    /// # Safety
+    ///
+    /// This doesn't take into account padding, so if `T` has padding
+    /// the result will be `undef`, which cannot be exposed to safe code.
```

Comment:
Are you sure padding will return an `undef` result and not just cause UB immediately?

Reply:
I was sure when I wrote that comment, but then I changed what it does 😅 

Treating it as branching on the bytes, and thus being UB for anything not fully-initialized, definitely sounds like the better semantic for it.